vpr/src/place/simpleRL_move_generator.h

```diff
@@ -65,20 +69,24 @@ class EpsilonGreedyAgent : public KArmedBanditAgent {
  */
 class SoftmaxAgent : public KArmedBanditAgent {
   public:
-    SoftmaxAgent(size_t num_actions);
+    SoftmaxAgent(size_t num_moves);
+    SoftmaxAgent(size_t num_moves, size_t num_types);
     ~SoftmaxAgent();
 
     //void process_outcome(double reward, std::string reward_fun) override; //Updates the agent based on the reward of the last proposed action
-    size_t propose_action() override; //Returns the type of the next action the agent wishes to perform
+    t_propose_action propose_action() override; //Returns the type of the next action as well as the block type the agent wishes to perform
 
   public:
     void set_action_prob();
+    void set_block_ratio();
```

Comment:
comment each function (all new ones for sure, existing if you know what they do). Can be short.